### PR TITLE
[mesos] Add support for NUM_MINION to mesos/docker cluster

### PR DESCRIPTION
**Pull request by sttts**
``` bash
$ NUM_MINION=4 kube-up.sh
```

### CODE REVIEW

File: cluster/mesos/docker/docker-compose.yml [hunk 1]
```diff
@@ -26,11 +26,11 @@ mesosmaster1:
...
   - etcd
   - "ambassador:apiserver"
-mesosslave1:
-  hostname: mesosslave1
+mesosslave:
+  hostname: mesosslave
   privileged: true
   image: mesosphere/mesos-slave-dind:0.23.0-1.0.ubuntu1404.docker181
-  entrypoint: [ "bash", "-c", "wrapdocker mesos-slave --hostname=$(getent hosts mesosslave1 | cut -d' ' -f1 | sort -u | tail -1)" ]
+  entrypoint: [ "bash", "-xc", "env | sort; NAME=$(cut -f2 -d/ <<<$MESOSMASTER1_NAME); N=${NAME##*_}; rm -rf /var/tmp/mesos/$N/*; MESOS_WORK_DIR=/var/tmp/mesos/$N DOCKER_NETWORK_OFFSET=0.0.$N.0 MESOS_ATTRIBUTES=rack:$N\\;gen:201$N wrapdocker mesos-slave --hostname=$(getent hosts mesosslave | cut -d' ' -f1 | sort -u | tail -1)" ]
```
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: use mesos cmd args, instead of env vars here

File: cluster/mesos/docker/util.sh [hunk 2]
```diff
@@ -256,7 +254,7 @@ function cluster::mesos::docker::init_auth {
 function kube-up {
   # Nuke old mesos-slave workspaces
   for ((i=1; i <= NUM_MINIONS; i++)) do
```
Comment (karlkfi) [inline cluster/mesos/docker/util.sh]: likely want to nuke all the dirs, not just the ones we're currently configured for. ex: downsizing

**Comment by sttts** [comment]
@karlkfi addressed your comments. ptal.

**Comment by sttts** [comment]
@karlkfi ptal

File: cluster/mesos/docker/docker-compose.yml [hunk 3]
```diff
@@ -26,34 +26,22 @@ mesosmaster1:
...
-  privileged: true
-  image: mesosphere/mesos-slave-dind:0.23.0-1.0.ubuntu1404.docker181
-  entrypoint: [ "bash", "-c", "wrapdocker mesos-slave --hostname=$(getent hosts mesosslave2 | cut -d' ' -f1 | sort -u | tail -1)" ]
+  entrypoint:
+  - bash
+  - -xc
+  - >
+    NAME=$(cut -f2 -d/ <<<$MESOSMASTER1_NAME) &&
+    N=${NAME##*_} &&
+    rm -rf /var/tmp/mesos/$N/* &&
```
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: I don't think we want to be wiping out the workspace here. That removes the possibility of starting a stopped cluster without destroying and recreating it. I know we don't support that yet, but I've got that working in a different mesos dind cluster. So I want to move in that direction. Basically the workspaces should be cleaned externally. They're mounted in, so the host technically "owns" them.

File: cluster/mesos/docker/docker-compose.yml [hunk 4]
```diff
@@ -26,34 +26,22 @@ mesosmaster1:
...
+    DOCKER_NETWORK_OFFSET=0.0.$N.0
```
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: does this work if N starts with zero?

File: cluster/mesos/docker/docker-compose.yml [hunk 5]
```diff
@@ -26,34 +26,22 @@ mesosmaster1:
...
+    exec wrapdocker mesos-slave
+    --work_dir="/var/tmp/mesos/$N"
+    --attributes=rack:$N\\;gen:201$N
```
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: are these just labels so we can pass an e2e test or is there some semantic meaning to them?

File: cluster/mesos/docker/util.sh [hunk 6]
```diff
@@ -288,6 +283,8 @@ function kube-up {
 
   echo "Starting ${KUBERNETES_PROVIDER} cluster" 1>&2
   cluster::mesos::docker::docker_compose up -d
+  echo "Scaling ${KUBERNETES_PROVIDER} cluster to ${NUM_MINIONS} slaves"
+  cluster::mesos::docker::docker_compose scale mesosslave=${NUM_MINIONS}
```
Comment (karlkfi) [inline cluster/mesos/docker/util.sh]: bugs me that scale can't be done in the yaml, but it looks like that conversation has been going on for more than a year in the fig/compose repo.

Comment (sttts) [inline cluster/mesos/docker/util.sh]: Yeah, have read the issue as well. That's a bit odd indeed.

File: cluster/mesos/docker/docker-compose.yml (hunk 5 above)
Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: just labels which are (will be) used in e2e tests.

File: cluster/mesos/docker/docker-compose.yml (hunk 4 above)
Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: it doesn't. it's the number of the docker-compose scale instance.

File: cluster/mesos/docker/docker-compose.yml (hunk 3 above)
Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: If we don't, the `docker-compose scale` will issue a `docker start` on the slave and this breaks the mesos-slave. Hence, we have to clean up.

File: cluster/mesos/docker/docker-compose.yml (hunk 5 above)
Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: In fact, our own e2e test in mesos.go.

File: cluster/mesos/docker/docker-compose.yml (hunk 3 above)
Comment (karlkfi) [inline cluster/mesos/docker/docker-compose.yml]: How is that bad, if we nuked the whole mount for prior to doing the docker-compose up?

Comment (sttts) [inline cluster/mesos/docker/docker-compose.yml]: Tested again. I think I had an old, stopped container before. With a fresh docker or after kube-down it works fine without the rm.

**Comment by sttts** [comment]
Rebased to fix e2e tests.

**Comment by sttts** [comment]
@karlkfi ptal

**Pull request merged**